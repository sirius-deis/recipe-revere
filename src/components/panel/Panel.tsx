import { FC, PropsWithChildren } from "react";
import styles from "./Panel.module.css";

interface PanelProps {
  withBorder?: boolean;
  withShadow?: boolean;
  centered?: boolean;
  [x: string]: any;
}

const Panel: FC<PropsWithChildren & PanelProps> = ({
  children,
  withBorder,
  withShadow,
  centered,
  ...rest
}) => {
  return (
    <article
      className={`${styles.panel}${withBorder ? ` ${styles.bordered}` : ""}${
        withShadow ? ` ${styles.shadowed}` : ""
      }${centered ? ` ${styles.centered}` : ""}`}
      data-testid="panel"
      {...rest}
    >
      {children}
    </article>
  );
};

export default Panel;
