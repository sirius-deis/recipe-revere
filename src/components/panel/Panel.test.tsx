import { render, screen } from "@testing-library/react";
import Panel from "./Panel";

const dummyText =
  "Lorem ipsum dolor sit amet consectetur adipisicing elit. Vero, consectetur maxime eius consequuntur illo eaque sit nam exercitationem totam dolore harum eos modi assumenda aperiam minus maiores. Eaque, accusamus perferendis.";
const dummyHtml = (
  <div>
    <ul>
      <li>list item 1</li>
      <li>list item 2</li>
    </ul>
  </div>
);
describe("Panel component", () => {
  it("should match snapshot", () => {
    const { container: container1 } = render(<Panel>dummyText</Panel>);
    const { container: container2 } = render(<Panel>dummyHtml</Panel>);
    expect(container1).toMatchSnapshot();
    expect(container2).toMatchSnapshot();
  });
  it("should render component with proper text", () => {
    render(<Panel>{dummyText}</Panel>);
    expect(screen.getByText(dummyText)).toBeInTheDocument();
  });
  it("should render component with proper html inside it", () => {
    render(<Panel>{dummyHtml}</Panel>);
    expect(screen.getByRole("list")).toBeInTheDocument();
    expect(screen.getByText("list item 1")).toBeInTheDocument();
    expect(screen.getByText("list item 2")).toBeInTheDocument();
  });
  it("should render component with correct props set without passing ant props", () => {
    render(<Panel>{dummyText}</Panel>);
    expect(screen.getByTestId("panel")).not.toHaveClass("bordered");
    expect(screen.getByTestId("panel")).not.toHaveClass("shadowed");
  });
  it("should render component with correct props set without passing ant props", () => {
    render(
      <Panel withBorder withShadow>
        {dummyText}
      </Panel>
    );
    expect(screen.getByTestId("panel")).toHaveClass("bordered");
    expect(screen.getByTestId("panel")).toHaveClass("shadowed");
  });
});
