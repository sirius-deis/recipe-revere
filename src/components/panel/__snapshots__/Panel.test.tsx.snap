// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel component should match snapshot 1`] = `
<div>
  <article
    class="panel"
    data-testid="panel"
  >
    dummyText
  </article>
</div>
`;

exports[`Panel component should match snapshot 2`] = `
<div>
  <article
    class="panel"
    data-testid="panel"
  >
    dummyHtml
  </article>
</div>
`;
